diff --git a/logread.c b/logread.c
index 911fa13..c526081 100644
--- a/logread.c
+++ b/logread.c
@@ -101,33 +101,32 @@ void execute_records(Record *Rarr, unsigned int num_records, struct slisthead *h
 }
 
 void print_time(Record *Rarr, unsigned int num_records, char *name) {
-  //TODO code this
-  //printf("Called print_time\n");
-  unsigned long min_time = 0;
-  unsigned long max_time = 0;
+  unsigned long total_time = 0;
+  unsigned long last_arrival = 0;
+  int inside = 0;
   int found = 0;
 
   //iterate through Record array
   for (unsigned int i = 0; i < num_records; i++) {
     //check if the record belongs to the person we are looking for
-    if (strcmp(Rarr[i].name, name) == 0) {
-      unsigned long current_time = (unsigned long)Rarr[i].timestamp;
-
-      if (!found) {
-        //if we haven't found this person yet, initialize with the first timestamp
-        min_time = current_time;
-        max_time = current_time;
-        found = 1;
-      } else {
-        //update the min and max
-        if (current_time < min_time) min_time = current_time;
-        if (current_time > max_time) max_time = current_time;
+    if (strcmp(Rarr[i].name, name) == 0 && Rarr[i].room == -1) {
+      //unsigned long current_time = (unsigned long)Rarr[i].timestamp;
+      found = 1;
+      //if person arrived at gallery
+      if (Rarr[i].actionType == Arrived) {
+        last_arrival = (unsigned long)Rarr[i].timestamp;
+        inside = 1;
+      }
+      //if person left and was previously inside the gallery
+      else if (Rarr[i].actionType == Left && inside) {
+        total_time += ((unsigned long)Rarr[i].timestamp - last_arrival);
+        inside = 0;
       }
     }
   }
 
   if (found) {
-    printf("%lu", max_time - min_time);
+    printf("%lu", total_time);
   }
 
   return;
