diff --git a/logread.c b/logread.c
index 911fa13..73952c8 100644
--- a/logread.c
+++ b/logread.c
@@ -339,6 +339,12 @@ int main(int argc, char *argv[]) {
     logpath = argv[optind];
   }
 
+  //check for additional, illegal arguments
+  if(optind < argc) {
+    printf("invalid\n");
+    return 255;
+  }
+
   //validate data we collected from commandline
   if (!token || !logpath) {
     printf("invalid\n");
